instance/schemas/theming.py

```diff
@@ -19,74 +19,122 @@
-link_color = '#126f9a'
-# openedx also uses white by default
-header_bg_color = '#ffffff'
-# openedx also uses white by default
-footer_bg_color = '#ffffff'
+    return {
+        "$ref": "#/definitions/{}".format(definition_name)
+    }
 
-text_color_options = Or('light', 'dark', color)
-main_color_options = Or('accent', 'main', color)
 
-button_color_schema = {
-    Optional('main'): main_color_options,
-    Optional('text'): text_color_options,
-    Optional('line'): nullable(main_color_options),
-    Optional('hover-main'): main_color_options,
-    Optional('hover-text'): text_color_options,
-    Optional('hover-line'): nullable(main_color_options),
+color_schema = {
+    "type": "string",
+    "pattern": color_regex,
 }
 
-theme_schema_v0 = Schema({
-    'version': 0,
-    # This is used as the primary color in your theme palette. It is used as filler for buttons.
-    Optional('main_color', default=main_color): color,
-    # This is used as the color for clickable links on your instance.
-    Optional('link_color', default=link_color): color,
-    # Header background color: Used as the background color for the top bar.
-    Optional('header_bg_color', default=header_bg_color): color,
-    # Footer background color: Used as the background color for the footer.
-    Optional('footer_bg_color', default=header_bg_color): color,
-})
+theme_schema_v0 = {
+    "definitions": {
+        "color": color_schema
+    },
+    "type": "object",
+    "properties": {
+        "version": {"type": "number", "enum": [0, ]},
+        "main_color": ref("color"),
+        "link_color": ref("color"),
+        "header_bg_color": ref("color"),
+        "footer_bg_color": ref("color"),
+    },
+    "required": ["version", "main_color", "link_color", "header_bg_color", "footer_bg_color"]
+}
 
-theme_schema_v1 = Schema({
-    'version': 1,
-    Optional('theme', default='light'): Or('light', 'dark'),
-    'colors': {
-        Optional('main'): nullable(color),
-        Optional('accent'): nullable(color),
-        Optional('buttons'): {
-            'primary': button_color_schema,
-            'secondary': button_color_schema,
+theme_schema_v1 = {
+    "definitions": {
+        "color": color_schema,
+        "btn": {
+            "type": "object",
+            "properties": {
+                "bg": ref("color"),
+                "fg": ref("color"),
+                "border": ref("color"),
+                "hover-bg": ref("color"),
+                "hover-fg": ref("color"),
+                "hover-border": ref("color"),
+            },
+            "required": ["bg", "fg", "border", "hover-bg", "hover-fg", "hover-border"]
+        },
+        "flag": {
+            "enum": [True, ]
+        },
+        "sacondary_nav": {
+            "type": "object",
+            "properties": {
+                "color": ref("color"),
+                "border-bottom": ref("color"),
+            }
         }
-    }
-})
+    },
+    "type": "object",
+    "properties": {
```

Comment:
@lgp171188 How does the mapping of unassigned variables work here?
Let's say the user only set `main_color`, `accent_color` and `light` in the theme options, is the validation going to fail?

Reply:
@giovannicimolin, as discussed in the meeting we had yesterday, whatever values are stored in the database will be rendered. Now, what the client has to send and what has to be done in the API handler before save is still TBD.